File: weapp/miniprogram/api/agreement.ts
import { request } from '../utils/request'

export interface AgreementBrief {
  type: string
  title: string
  version: string
  published_on: string
}

export interface AgreementDetail {
  id: number
  type: string
  title: string
  content: string
  version: string
  published_on: string
  is_active: boolean
  created_at: string
  updated_at: string
}

export const AgreementService = {
  /**
   * 获取活跃协议列表
   */
  async listActiveAgreements(): Promise<AgreementBrief[]> {
    return request({ url: '/v1/agreements', method: 'GET' })
  },

  /**
   * 获取协议详情
   * @param type 协议类型
   */
  async getAgreement(type: string): Promise<AgreementDetail> {
    return request({ url: `/v1/agreements/${type}`, method: 'GET' })
  }
}

File: locallife/api/agreement.go
package api

import (
	"net/http"

	"github.com/gin-gonic/gin"
)

type listActiveAgreementsResponse struct {
	Type        string `json:"type"`
	Title       string `json:"title"`
	Version     string `json:"version"`
	PublishedOn string `json:"published_on"`
}

// @Summary List active agreements
// @Description get the list of all active agreements
// @Tags agreements
// @Accept json
// @Produce json
// @Success 200 {array} listActiveAgreementsResponse
// @Router /v1/agreements [get]
func (server *Server) listAgreements(ctx *gin.Context) {
	agreements, err := server.store.ListActiveAgreements(ctx)
	if err != nil {
		ctx.JSON(http.StatusInternalServerError, internalError(ctx, err))
		return
	}

	response := make([]listActiveAgreementsResponse, len(agreements))
	for i, a := range agreements {
		response[i] = listActiveAgreementsResponse{
			Type:        a.Type,
			Title:       a.Title,
			Version:     a.Version,
			PublishedOn: a.PublishedOn.Time.Format("2006-01-02"),
		}
	}

	ctx.JSON(http.StatusOK, response)
}

// @Summary Get active agreement by type
// @Description get the content of an active agreement by its type
// @Tags agreements
// @Accept json
// @Produce json
// @Param type path string true "Agreement Type"
// @Success 200 {object} db.Agreement
// @Router /v1/agreements/{type} [get]
func (server *Server) getAgreement(ctx *gin.Context) {
	agreementType := ctx.Param("type")

	agreement, err := server.store.GetActiveAgreementByType(ctx, agreementType)
	if err != nil {
		ctx.JSON(http.StatusInternalServerError, internalError(ctx, err))
		return
	}

	ctx.JSON(http.StatusOK, agreement)
}
